src/sagemaker/workflow/airflow.py

```diff
@@ -0,0 +1,137 @@
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import os
+
+from sagemaker.estimator import Framework
+from sagemaker.amazon.amazon_estimator import AmazonAlgorithmEstimatorBase, RecordSet
+from sagemaker.job import _Job
+from sagemaker.utils import base_name_from_image, airflow_name_from_base
+from sagemaker.model import DIR_PARAM_NAME, SCRIPT_PARAM_NAME, CLOUDWATCH_METRICS_PARAM_NAME, \
+    CONTAINER_LOG_LEVEL_PARAM_NAME, JOB_NAME_PARAM_NAME, SAGEMAKER_REGION_PARAM_NAME
+
+
+def prepare_framework(estimator, s3_operations, default_bucket):
+    """
+    Prepare S3 operations (specify where to upload source_dir) and environment variables
+        related to framework.
+
+    Args:
+        estimator: The framework estimator to get information from and update.
+        s3_operations: The dict to specify s3 operations (upload source_dir).
+        default_bucket: The default bucket to use in training.
+    """
+    bucket = estimator.code_location if estimator.code_location else default_bucket
+    key = '{}/source/sourcedir.tar.gz'.format(estimator._current_job_name)
+    script = os.path.basename(estimator.entry_point)
+    if estimator.source_dir and estimator.source_dir.lower().startswith('s3://'):
+        code_dir = estimator.source_dir
+    else:
+        code_dir = 's3://{}/{}'.format(bucket, key)
+        s3_operations['S3Upload'] = [{
+            'Path': estimator.source_dir or script,
+            'Bucket': bucket,
+            'Key': key,
+            'Tar': True
+        }]
+    estimator._hyperparameters[DIR_PARAM_NAME] = code_dir
+    estimator._hyperparameters[SCRIPT_PARAM_NAME] = script
+    estimator._hyperparameters[CLOUDWATCH_METRICS_PARAM_NAME] = estimator.enable_cloudwatch_metrics
+    estimator._hyperparameters[CONTAINER_LOG_LEVEL_PARAM_NAME] = estimator.container_log_level
+    estimator._hyperparameters[JOB_NAME_PARAM_NAME] = estimator._current_job_name
+    estimator._hyperparameters[SAGEMAKER_REGION_PARAM_NAME] = estimator.sagemaker_session.boto_region_name
+
+
+def prepare_amazon_algorithm_estimator(estimator, inputs):
+    """
+    Set up amazon algorithm estimator, adding the required feature_dim hyperparameter from training data.
+    Args:
+        estimator: The Amazon algorithm estimator to get information from and update.
+        inputs: The training data, must be in RecordSet format.
+    """
+    if isinstance(inputs, list):
+        for record in inputs:
+            if isinstance(record, RecordSet) and record.channel == 'train':
+                estimator.feature_dim = record.feature_dim
+                break
+    elif isinstance(inputs, RecordSet):
+        estimator.feature_dim = inputs.feature_dim
+    else:
+        raise TypeError('The training data of Amazon Algorithm estimator is not represented by RecordSet.')
+
+
+def get_training_config(estimator, inputs=None, job_name=None):
```

Comment:
Don't use **get**  or **set** in Python. What about ```training_config```?

Reply:
Updated.